src/change_stream.ts

```diff
@@ -547,17 +551,17 @@ const CHANGE_STREAM_EVENTS = [
 
 function setIsEmitter<TSchema>(changeStream: ChangeStream<TSchema>): void {
   if (changeStream[kMode] === 'iterator') {
-    throw new MongoDriverError(
-      'Cannot use ChangeStream as an EventEmitter after using as an iterator'
+    throw new MongoChangeStreamError(
+      'ChangeStream cannot be used as an EventEmitter after being used as an iterator'
```

Comment:
On further consideration, the error thrown here and in `setIsIterator` should both be `MongoAPIError` instances.

While these functions are internal, the errors result directly from the user violating the change stream API. When a change stream is created, it should either be used as an iterator _or_ as an event emitter (NODE-1797). The user has to make the decision to pick which API they want to use.

Reply:
The TODO in L551 references NODE-3484, which doesn't make sense for this ticket, please update to TODO-3485